### ISSUE TITLE
Segmentation fault on some pools without describing a reason of that fault

### ISSUE THREAD

**Issue by wwarlock** [Lolliedieb/lolMiner-releases#972]
Trying to develop the pool and chosen lolMiner to test it.
lolMiner is sending the Authorize request and falling right after it without waiting.
What answer lolMiner is waiting from pool?

```bash
Setup Miner... 
OpenCL driver detected.
Number of OpenCL supported GPUs: 2 
Cuda driver detected.
Number of Cuda supported GPUs: 1 
Device 0: 
    Name:    GeForce GTX 1050 
    Address: 1:0 
    Vendor:  NVIDIA Corporation
    Drivers: Cuda
    Memory:  2002 MByte 
    Active:  true (Selected Algorithm: TON (Sha 256)) 
Device 1: 
    Name:    Intel(R) Graphics Gen9 [0x591b] 
    Address: 4294967295:4294967295 
    Vendor:  Intel(R) Corporation
    Drivers: OpenCL
    Memory:  25620 MByte 
    Active:  false (not selected) 

Connecting to pool... 
Ton pool connection mode: toncoinpool.io stratum (--ton-mode 3) 
Connected to 127.0.0.1(127.0.0.1):3333  (TLS disabled)
Subscribed to stratum server. Pool difficulty: infT 
Authorized worker: EQAhy_my_ton_wallet_number_here_g 
./mine_ton.sh: line 27: 2973605 Segmentation fault      (core dumped) ./lolMiner --algo TON --pool $POOL --user $WALLET --ton-mode 3 $@

```

**Comment by jgonzis** [comment]
> Trying to develop the pool and chosen lolMiner to test it. lolMiner is sending the Authorize request and falling right after it without waiting. What answer lolMiner is waiting from pool?
> 
> ```shell
> Setup Miner... 
> OpenCL driver detected.
> Number of OpenCL supported GPUs: 2 
> Cuda driver detected.
> Number of Cuda supported GPUs: 1 
> Device 0: 
>     Name:    GeForce GTX 1050 
>     Address: 1:0 
>     Vendor:  NVIDIA Corporation
>     Drivers: Cuda
>     Memory:  2002 MByte 
>     Active:  true (Selected Algorithm: TON (Sha 256)) 
> Device 1: 
>     Name:    Intel(R) Graphics Gen9 [0x591b] 
>     Address: 4294967295:4294967295 
>     Vendor:  Intel(R) Corporation
>     Drivers: OpenCL
>     Memory:  25620 MByte 
>     Active:  false (not selected) 
> 
> Connecting to pool... 
> Ton pool connection mode: toncoinpool.io stratum (--ton-mode 3) 
> Connected to 127.0.0.1(127.0.0.1):3333  (TLS disabled)
> Subscribed to stratum server. Pool difficulty: infT 
> Authorized worker: EQAhy_my_ton_wallet_number_here_g 
> ./mine_ton.sh: line 27: 2973605 Segmentation fault      (core dumped) ./lolMiner --algo TON --pool $POOL --user $WALLET --ton-mode 3 $@
> ```

 is it trying to  connect to 127.0.0.1??

**Comment by wwarlock** [comment]
Yes, and it sending `subscribe` and `authorize` requests before it die.

**Comment by wwarlock** [comment]
I checked the miner with pool `wss://pplns.toncoinpool.io/stratum` and still having almost the same error
```
Connecting to pool... 
Ton pool connection mode: toncoinpool.io stratum (--ton-mode 3) 
./mine_ton.sh: line 30: 303039 Segmentation fault      (core dumped) ./lolMiner --log on --algo TON --pool $POOL --user $WALLET --ton-mode 3 $@
```

But as I found the `--tls off` key is slightly changed the output 
```
Connecting to pool... 
Ton pool connection mode: toncoinpool.io stratum (--ton-mode 3) 
DNS over HTTPS resolve succeeded 
Connected to wss://pplns.toncoinpool.io/stratum(104.22.15.214):443  (TLS enabled)
./mine_ton.sh: line 28: 303611 Segmentation fault      (core dumped) ./lolMiner --log on --tls off --algo TON --pool $POOL --user $WALLET --ton-mode 3 $@
```

In any case here no subscribe or authorize requests.
Can't segmentation fault be caused by some security (SSL/TLS) troubles?

**Comment by jgonzis** [comment]
All these seg-fault should be solved with last release 1.46
